File: portal/portal/App_Code/DAO/MidiaDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;

namespace portal.App_Code.DAO
{
    public class MidiaDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public Midia carregar(long pCodigo)
        {
            Midia obj = new Midia();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Midia where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Url = registros["url"].ToString();
                    obj.Nome = registros["nome"].ToString();
                    obj.Data = registros["data_publicado"].ToString();
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<Midia> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<Midia> lista = new List<Midia>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from Midia where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    Midia obj = new Midia();
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Url = registros["url"].ToString();
                    obj.Nome = registros["nome"].ToString();
                    obj.Data = registros["data_publicado"].ToString();
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(Midia obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into Midia(nome, url, data_publicado) value('{0}','{1}',sysdate)";
                    sql = String.Format(sql, obj.Nome, obj.Url);
                }
                else
                {
                    sql = "update Midia set nome='{0}', url='{1}' where codigo={2}";
                    sql = String.Format(sql, obj.Nome, obj.Url,  obj.Codigo);
                }

                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(Midia obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from Midia where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/MidiaDAO.php
<?php
namespace portal\App_Code\DAO {
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\Midia;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);     

    try{
        require_once("/../Midia.php");        
        require_once("/../../connection.php");  
    }catch(Exception $e){
        require_once (dirname(__file__, 2) . "\Midia.php");
        require_once (dirname(__file__, 3) . "\connection.php");   
    }

class MidiaDAO{

//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
function inserirMidia($midia){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into midia (nome, url, data_publicado) values (' " .
                 $midia->getNome() . " ',' " .
                 $midia->getURL() . " ',' " .
                 $midia->getData() . " ')";
                 mysqli_query($con,$sql);
                 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
}

//ALTERAR--------------------------------------------------------------------------------------------------------------------------------------
function alterarMidia($midia){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update midia set " . 
                "nome = ' " . $midia->getNome() . "', " .
                "url = ' " . $midia->getURL() . "', " .
                "data_publicado = '" . $midia->getData() . "' ";
                mysqli_query($con,$sql);
                echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }

//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
    function carregarMidia ($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from midia where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
                        $Mid = new Midia();
                        $Mid->setMidia(($carga["codigo"]),
                                       ($carga["nome"]),
                                       ($carga["url"]),
                                       ($carga["data_publicado"]));
                        mysqli_close($con);
                        return $Mid;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;  
            }
        mysqli_close($con);
    }

//CARREGAR LISTA DE MÍDIAS-----------------------------------------------------------------------------------------------------------------------
    function carregarListaMidia($qtdOpcoes, $ordem){
           $sql = "select * from midia where "; 
        
             for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++){
                 $sql = $sql . $qtdOpcoes[$i];
                    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    } 
            }
        
            $sql = $sql . " order by " . $ordem;
        
        try{
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);                
            if($carga = mysqli_fetch_array($gargantula)){
							  $ListaMidia = [];
                        while($carga = mysqli_fetch_array($gargantula)){
                              $midia = new portal\App_Code\Midia();
                              $midia->setMidia(($carga["codigo"]),
                                                 ($carga["nome"]),
                                                 ($carga["url"]),
                                                 ($carga["data_publicado"]));
                              $ListaMidia[] = $midia;
                         }
                        mysqli_close($con);
                        return $ListaMidia;
                    }else{
                        echo "Registros não encontrados!";
                    }
        }catch(Exception $e){
                echo "Erro de busca: " . $e;           
            }
mysqli_close($con);        
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarMidia($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from midia where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
                echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;            
            }
        mysqli_close($con);
    }
  }
}
?>